src/main/java/com/arpnetworking/metrics/mad/parsers/JsonToRecordParser.java

```diff
@@ -237,6 +241,46 @@ public Record parse(final byte[] data) throws ParsingException {
     }
 
     // NOTE: Package private for testing
+    /* package private */com.arpnetworking.metrics.mad.model.Record parseV2gLogLine(final JsonNode jsonNode)
+            throws JsonProcessingException {
+
+        final Version2g model = OBJECT_MAPPER.treeToValue(jsonNode, Version2g.class);
+        final Version2g.Annotations annotations = model.getAnnotations();
+        final ImmutableMap.Builder<String, Metric> variables = ImmutableMap.builder();
+
+        putVariablesVersion2g(model.getTimers(), MetricType.TIMER, variables);
+        putVariablesVersion2g(model.getCounters(), MetricType.COUNTER, variables);
+        putVariablesVersion2g(model.getGauges(), MetricType.GAUGE, variables);
+
+        // NOTE: We are mapping from the underscore prefixed annotations to non-prefixed ones.
```

Comment:
I'm perfectly fine with having non-underscored and having them be directly in dimensions.